canonical/maas-ui
# feat: tag management panel MAASENG-1427

## Done

- feat: tag management panel MAASENG-1427
  - remove .l-aside visibility workaround 
  - refactor AppSidePanel
  - update node action menu dropdown styles
  - remove eslint-disable-next-line cypress/no-force comments as we disable the rule globally due to vanillaframework implementation of form elements

Details on the design: https://warthogs.atlassian.net/browse/MAASENG-1422

## QA


### QA steps

- Go to machine list page
- Select a machine
- Click categorise -> tag
- type a name of a new tag in the search field
- add a new tag
- Click save
- Ensure the machine has been updated correctly
---
- go to machine details page
- Click categorise -> tag
- type a name of a new tag in the search field
- add a new tag
- Click save
- Ensure the machine has been updated correctly
- ---
- Go to devices page
- Press "Add"
- Click "Controllers" in the navigation menu
- Verify the side panel has been closed 

## Fixes

Fixes: https://warthogs.atlassian.net/browse/MAASENG-1427

## Screenshots
#### After
![image](https://github.com/canonical/maas-ui/assets/7452681/97cbcc3d-ae8d-4a1a-935f-e209e1d44d5a)
![image](https://github.com/canonical/maas-ui/assets/7452681/3edafa3e-e31f-4585-85a3-d519377a2898)

### Node action menu - mobile
#### Before
<img src="https://github.com/canonical/maas-ui/assets/7452681/9728d232-c4b0-4e48-b8ef-0bad0a987f26" width="600" />

#### After
<img src="https://github.com/canonical/maas-ui/assets/7452681/d0246451-a74f-4e6a-9786-c09c147465f1" width="600" />

### Node action menu - desktop
#### Before
<img src="https://github.com/canonical/maas-ui/assets/7452681/dcf4a94c-8ec6-4539-b1f7-31e17771461e" width="600" />

#### After
<img src="https://github.com/canonical/maas-ui/assets/7452681/dbe41cd9-6acd-4d53-97d7-f61ea2c3c0a4" width="600" />
